libs/nestjs-admin/src/views/widgets/simple-array.njk

```diff
@@ -2,21 +2,35 @@
 
 {% block input %}
   {% macro field(name, value = '') %}
-    <div>
-      <input
-        id="{{ name }}"
-        name="{{ name }}"
-        value="{{ value }}"
-      />
-      <button type="button">X</button>
-    </div>
+    <input
+      id="{{ name }}"
+      name="{{ name }}"
+      value="{{ value }}"
+    />
+    <button type="button" onclick="removeInput(this)">X</button>
   {% endmacro %}
 
-  {% for value in column.getEntityValue(entity) %}
-    {% set fieldName = column.propertyName + "["+ loop.index0+ "]" %}
-    {{ field(fieldName, value) }}
-  {% endfor %}
+  <div id="values-{{ column.propertyName }}">
+    {% for value in column.getEntityValue(entity) %}
+    <div class="input-{{ column.propertyName }}">
+      {{ field(column.propertyName, value) }}
+    </div>
+    {% endfor %}
+  </div>
+  <button type="button" onclick="addInput()">+</button>
+{% endblock %}
 
-  {% set fieldName = column.propertyName + "["+ "3" + "]" %}
-  {# <button type="button" onclick="this.parentElement.append('{{field(fieldName)|forceescape}}')">+</button> #}
+{% block script %}
+  <script>
+    function addInput() {
+      const inputContainer = document.getElementById("values-{{ column.propertyName }}");
+      const newInput = "{{ field(column.propertyName) | replace("\n", "") | replace("\"", "\\\"") }}";
+      const node = document.createElement("div");
+      node.innerHTML = newInput;
+      inputContainer.appendChild(node)
+    }
+    function removeInput(self) {
+      self.closest('.input-{{ column.propertyName }}').remove()
+    }
+  </script>
```

Comment:
Instead of this block `script` here, that would get included several times if you have several array fields, could you add it to the `base.njk`? Or maybe even better, create a `form.njk` and put it there, having `change` and `add` extending it